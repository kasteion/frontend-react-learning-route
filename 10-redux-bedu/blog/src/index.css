body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

.margen {
  margin: 100px;
}

.tabla {
  width: 100%;
  text-align: left;
}

.tabla td {
  padding-top: 2vh;
  padding-bottom: 2vh;
}

#menu {
  background-color: #253a46;
  padding: 20px;
  font-size: 20px;
}

#menu a {
  color: white;
  padding-right: 50px;
  text-decoration: none;
}

.pub-title {
  border-top: 1px solid black;
  cursor: pointer;
}

.contenedor_tareas {
  font-size: 20px;
  margin-left: 50px;
  margin-bottom: 70px;
}

.m_left {
  margin-left: 20px;
}
