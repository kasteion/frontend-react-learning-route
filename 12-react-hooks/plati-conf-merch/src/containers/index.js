export { default as Home } from "../containers/Home";
export { default as Checkout } from "../containers/Checkout";
export { default as Information } from "../containers/Information";
export { default as Payment } from "../containers/Payment";
export { default as Success } from "../containers/Success";
export { default as NotFound } from "../containers/NotFound";
