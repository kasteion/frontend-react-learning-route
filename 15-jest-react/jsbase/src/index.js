const cities = [
  "Ciudad de México",
  "Bogotá",
  "Lima",
  "Buenos Aires",
  "Guadalajara",
];

const randomCity = () => {
  const city = cities[Math.floor(Math.random() * cities.length)];
  return city;
};

const reverseString2 = (str) => {
  return new Promise((resolve, reject) => {
    if (!str) {
      reject(Error("Error"));
    }
    resolve(str.split("").reverse().join(""));
  });
};

module.exports = randomCity;
