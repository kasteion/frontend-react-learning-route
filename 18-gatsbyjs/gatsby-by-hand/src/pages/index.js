import React from "react";

const index = () => {
  return (
    <div>
      <h1>Gatsby by hand</h1>
    </div>
  );
};

export default index;
