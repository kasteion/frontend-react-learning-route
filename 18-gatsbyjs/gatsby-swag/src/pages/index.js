import * as React from "react"
import { graphql } from "gatsby"
// import { StaticImage } from "gatsby-plugin-image"

import Seo from "../components/seo"
import { Jumbo, Products } from "../components"

export const query = graphql`
  query GET_DESCRIPTION {
    allSite {
      edges {
        node {
          siteMetadata {
            description
          }
        }
      }
    }
  }
`

const IndexPage = ({ data }) => {
  // console.log(data.allSite.edges[0].node.siteMetadata.description)
  return (
    <>
      <Seo title="Home" />
      <Jumbo
        description={data.allSite.edges[0].node.siteMetadata.description}
      />
      <Products />
    </>
  )
}

export default IndexPage
