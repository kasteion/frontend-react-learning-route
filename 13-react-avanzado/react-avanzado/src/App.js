import React from 'react'
import { Router } from '@reach/router'
import Context from './Context'
import { GlobalStyle } from './styles/GlobalStyles'
import { Logo } from './components/Logo'
import { Home } from './pages/Home'
import { Detail } from './pages/Detail'
import { User } from './pages/User'
import { Favs } from './pages/Favs'
import { NotRegisteredUser } from './pages/NotRegisteredUser'
import { NavBar } from './components/NavBar'

export const App = () => {
  return (
    <>
      <GlobalStyle />
      <Logo />
      <Router>
        <Home path='/' />
        <Home path='/pet/:categoryId' />
        <Detail path='/detail/:detailId' />
      </Router>
      <Context.Consumer>
        {
          ({ isAuth }) => isAuth
            ? <Router><User path='/user' /><Favs path='/favs' /></Router>
            : <Router><NotRegisteredUser path='/user' /><NotRegisteredUser path='/favs' /></Router>
        }
      </Context.Consumer>
      <NavBar />
    </>
  )
}
