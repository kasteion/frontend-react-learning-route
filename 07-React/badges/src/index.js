import React from "react";
import ReactDOM from "react-dom";
import "bootstrap/dist/css/bootstrap.css";
import "./global.css";
//import Badges from "./pages/Badges";
import App from "./components/App";

ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  document.getElementById("root")
);
