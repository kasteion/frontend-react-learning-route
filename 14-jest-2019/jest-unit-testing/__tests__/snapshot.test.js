import { getCharacter } from "../src/snapshot";
import rick from "../src/rick.json";

describe("Es hora de las instantaneas", () => {
  test("Snapshot", () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });

  test("Siempre fallara la instantanea", () => {
    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 100),
    };
    expect(user).toMatchSnapshot();
  });

  test("Tenemos una excepciÃ³n para el snapshot", () => {
    const user = {
      id: Math.floor(Math.random() * 100),
      name: "Fredy",
    };
    expect(user).toMatchSnapshot({ id: expect.any(Number) });
  });
});
