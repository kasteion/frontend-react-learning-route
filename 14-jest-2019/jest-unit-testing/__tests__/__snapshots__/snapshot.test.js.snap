// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Siempre fallara la instantanea 1`] = `
Object {
  "createdAt": 2021-04-14T02:22:29.974Z,
  "id": 4,
}
`;

exports[`Es hora de las instantaneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instantaneas Tenemos una excepci√≥n para el snapshot 1`] = `
Object {
  "id": Any<Number>,
  "name": "Fredy",
}
`;
